### DISCUSSION TITLE
creating a static PVC from an existing CephFS volume

### DISCUSSION THREAD

**Discussion by duffrecords** [ceph/ceph-csi#4906] [Q&A] [answered]
Now that the `cephfs` volume type has been removed in Kubernetes 1.31, I'm trying to understand what the appropriate replacement would be.  I have an existing CephFS file system that I would like to attach to containers so they have access to some shared files.  I read the [instructions](https://github.com/ceph/ceph-csi/blob/devel/docs/static-pvc.md#cephfs-static-pvc) for creating a static PV and PVC from an existing subvolume but I don't have any of those--just a single CephFS volume.  Is it possible to do this without creating a subvolume?  The documentation seems to be focused on how to provision volumes of a fixed size but, as I understand it, CephFS scales with the underlying object store.  Would this mean I'd need to specify the size of the entire CephFS system?  I want to make sure that creating a static PVC is not going to try to provision a new block within the FS or delete existing data.

**Comment by Madhu-1** [comment, marked as answer]
>Is it possible to do this without creating a subvolume?

Yes it should be possible

>I want to make sure that creating a static PVC is not going to try to provision a new block within the FS or delete existing data.

Static PV doesn't create anything on the ceph system, it supports only mounting/umounting the PVC

**Comment by MondayCha** [comment]
> Now that the `cephfs` volume type has been removed in Kubernetes 1.31, I'm trying to understand what the appropriate replacement would be. I have an existing CephFS file system that I would like to attach to containers so they have access to some shared files. I read the [instructions](https://github.com/ceph/ceph-csi/blob/devel/docs/static-pvc.md#cephfs-static-pvc) for creating a static PV and PVC from an existing subvolume but I don't have any of those--just a single CephFS volume. Is it possible to do this without creating a subvolume? The documentation seems to be focused on how to provision volumes of a fixed size but, as I understand it, CephFS scales with the underlying object store. Would this mean I'd need to specify the size of the entire CephFS system? I want to make sure that creating a static PVC is not going to try to provision a new block within the FS or delete existing data.

Hello. May I ask if you have made any progress? I am in the process of migrating the super-computing cluster from Slurm to Kubernetes. For this purpose, I need to expose Ceph to the containers within Kubernetes through the Ceph CSI. 

However, the current user files are all saved under one Volume, without SubVolume. I sincerely hope to get your advice on this matter.

**Reply by Madhu-1** [reply]
Yes this is already supported, please give a try

**Reply by MondayCha** [reply]
> [!NOTE]
> - [ceph-csi/docs/static-pvc.md at devel · ceph/ceph-csi · GitHub](https://github.com/ceph/ceph-csi/blob/devel/docs/static-pvc.md)
> - [Create PV manually · rook/rook · Discussion #8861 · GitHub](https://github.com/rook/rook/discussions/8861)
> - [creating a static PVC from an existing CephFS volume · ceph/ceph-csi · Discussion #4906 · GitHub](https://github.com/ceph/ceph-csi/discussions/4906)
> - [Static PV using CephFS CSI · GitHub](https://gist.github.com/ShyamsundarR/1009b79966992470f588a277429ae2e1)
> - [cephfs static pv, rootPath is ignored by cephfs csi · Issue #1119 · ceph/ceph-csi · GitHub](https://github.com/ceph/ceph-csi/issues/1119)
> - [如何挂载一个已有的cephfs目录 · Issue #2746 · ceph/ceph-csi](https://github.com/ceph/ceph-csi/issues/2746)


Thank you very much for your reply. After reading the above materials, I have successfully imported the data from the previous supercomputing center without creating sub-volumes. Our users can thus have a smooth migration. Thanks again.

However, I noticed that the relevant solution has been available for four years. Combined with what [duffrecords](https://github.com/duffrecords) mentioned, there may be a wave of migrations after the k8s 1.31 version. Personally, I think it's necessary to add documentation on not creating sub-volumes. 

May I ask what your opinion is on this? If it's feasible, I'd be happy to submit a PR for this.

**Reply by Madhu-1** [reply]
with CSI we use subvolume concept per PVC, please feel to send a PR with required changes that need to be done for the filesystem/volume without subvolume as well.

**Reply by MondayCha** [reply]
Hi, I've submitted a PR #4997 and look forward to your feedback😊.

**Comment by duffrecords** [comment]
@MondayCha No, I haven't gotten to that step yet because I ended up resetting my Kubernetes cluster and starting over.  I'll post my findings once I get there.

**Comment by duffrecords** [comment]
I was able to get it working.  After doing some more reading, I now have a better understanding.  The capacity does not necessarily need to match the actual size of the CephFS volume.  It's more like a minimum requirement, i.e. an application requests a volume of a particular size and, if the storage class can satisfy the request, then the volume gets bound to the claim.  Also, as @Madhu-1 said, a subvolume is not required.  Subvolumes and groups are for separating the storage into logical units but you can just as easily specify `/` or any path within as the `rootPath`.

**Reply by MondayCha** [reply]
Thank you for your reply🤗. I have succeeded and submitted a PR for modifying the document (#4997).
